### ISSUE TITLE
Computing Cost of Non-local block

### ISSUE THREAD

**Issue by InstantWindy** [AlexHex7/Non-local_pytorch#20]
Hi！Is this module computationally heavy? When I add this module to my network, there are always CUDA out of memory errors. However, when I remove this module, it will work properly. I don't know why.Thanks

**Comment by AlexHex7** [comment]
Yes. It is only used in the top of Network. You can find the details in the paper called **Non-local Neural Networks**.
